src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java

```diff
@@ -84,10 +85,48 @@ public StreamingClientProperties(Map<String, String> connectorConfig) {
         Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_MAX_CLIENT_LAG));
     snowpipeStreamingMaxClientLag.ifPresent(
         overriddenValue -> {
-          LOGGER.info(
-              "Config is overridden for {}={}", SNOWPIPE_STREAMING_MAX_CLIENT_LAG, overriddenValue);
           parameterOverrides.put(MAX_CLIENT_LAG, String.format("%s second", overriddenValue));
         });
+    combineStreamingClientOverriddenProperties(connectorConfig);
+  }
+
+  private void combineStreamingClientOverriddenProperties(Map<String, String> connectorConfig) {
+    // MAX_CLIENT_LAG property can also be present in override map, in that case, we will honor the
+    // value
+    // provided in SNOWPIPE_STREAMING_MAX_CLIENT_LAG since it preceded and was released in earlier
+    // versions.
+    Optional<String> clientOverrideProperties =
+        Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_CLIENT_PARAMETER_OVERRIDE_MAP));
+    Map<String, String> clientOverridePropertiesMap = new HashMap<>();
+    clientOverrideProperties.ifPresent(
+        overriddenKeyValuePairs -> {
+          LOGGER.info(
+              "Overridden map provided for streaming client Properties: {}",
+              overriddenKeyValuePairs);
+          Map<String, String> overriddenPairs =
+              Utils.parseCommaSeparatedKeyValuePairs(overriddenKeyValuePairs);
+          overriddenPairs.forEach(
+              (overriddenKey, overriddenValue) -> {
+                if (overriddenKey.equalsIgnoreCase(MAX_CLIENT_LAG)) {
```

Comment:
because this is already its own property.

Reply:
We should probably call out somewhere in the docstring for the new override option that `MAX_CLIENT_LAG` is special cased.

Reply:
okay, I see you are talking about case sensitivity here. i thought you are talking about MAX_CLIENT_LAG's special `case` here

I checked ingest java, we will have to pass in all keys as lower case. https://github.com/snowflakedb/snowflake-ingest-java/blob/master/src/main/java/net/snowflake/ingest/utils/ParameterProvider.java#L204

I will pass in lowercase here.

Reply:
> We should probably call out somewhere in the docstring for the new override option that MAX_CLIENT_LAG is special cased.

This is not needed since we will always pass in a lowercase value in streaming provider regardless of what;s the input key.

Reply:
Oh I wasn't talking about upper or lower case, I was referring to that `MAX_CLIENT_LAG` is treated differently. We might want to tweak the documentation to reflect that and explain how that parameter and this new one impact each other.

Reply:
yes, I added javadocs with some examples.